noir-projects/noir-protocol-circuits/crates/types/src/utils/arrays.nr

```diff
@@ -83,6 +84,24 @@ pub fn array_concat<T, N, S, M>(array1: [T; N], array2: [T; S]) -> [T; M] where
     result
 }
 
+pub fn array_merge<T, N>(array1: [T; N], array2: [T; N]) -> [T; N] where T: Empty + Eq {
```

Comment:
do we need to assert that the non-empty length of array1 plus the non-empty length of array2 is less than or equal to N?

Reply:
I see we're checking in TS anyway so we should get the clear message.